app.ts

```diff
@@ -0,0 +1,301 @@
+		}
+	}, 250)
+})
+//display records from first to last number
+async function displayRecords(recordCount: number): Promise<void> {
+	$('#loader').show();
+	let calculatedRows = adjustRowsByScreenHeight();
+	const inputValue = $('#searchInput').val() as number;
+	const tbody = $("tbody");
+	if (calculatedRows === 0) {
+		globalVars.lastNumber = globalVars.firstNumber;
+	} else if (globalVars.firstNumber < 0) {
+		globalVars.firstNumber = 0;
+		globalVars.lastNumber = globalVars.firstNumber + (calculatedRows - 1);
+	} else {
+		globalVars.lastNumber = globalVars.firstNumber + (calculatedRows - 1);
+	}
+	if (globalVars.firstNumber === 0) {
+		$('.arrow-left').hide();
+	} else {
+		$('.arrow-left').show();
+	}
+	if (globalVars.lastNumber >= recordCount) {
+		$('.arrow-right').hide();
+		
+	} else {
+		$('.arrow-right').show();
+	}
+	// $('#page').empty().append(`Showing record: ${globalVars.firstNumber} - ${globalVars.lastNumber}`);
+	if (!isNaN(globalVars.firstNumber) && !isNaN(globalVars.lastNumber) && !isNaN(recordCount) && !isNaN(calculatedRows)) {
+		if (globalVars.lastNumber <= recordCount && globalVars.lastNumber >= 0) {
+			return globalVars
+				.fetchRecords(globalVars.firstNumber, globalVars.lastNumber)
+				.then((records) => {
+					$('#page').empty().append(`Showing record: ${globalVars.firstNumber} - ${globalVars.lastNumber}`);
+					$('#loader').hide();
+					tbody.empty();
+					for (const record of records) {
+						// creates row for each record
+						$("tbody").append(`<tr class="row"></tr>`);
+						const lastRow = $(".row:last");
+						for (const value of record) {
+							// assign each record to their column in a specified row
+							lastRow.append(`<td>${value}</td>`);
+						}
+						if (record.includes(inputValue)) {
+							// highlights the searched row
+							lastRow.css('background-color', '#DDC0B4');
+						}
+						$("tbody").append(lastRow);
+					}
+				})
+				.catch((err) => {
+					throw 'Error fetching records' + err;
+				})
+		} else {
+			globalVars.firstNumber = recordCount - (calculatedRows - 1);
+			globalVars.lastNumber = recordCount;
+			return globalVars
+				.fetchRecords(globalVars.firstNumber, recordCount)
+				.then((records) => {
+					$('#page').empty().append(`Showing record: ${globalVars.firstNumber} - ${globalVars.lastNumber}`);
+					$('#loader').hide();
+					tbody.empty();
+					for (const record of records) {
+						// creates row for each record
+						$("tbody").append(`<tr class="row"></tr>`);
+						const lastRow = $(".row:last");
+						for (const value of record) {
+							//assign each record to their column in a specified row
+							lastRow.append(`<td>${value}</td>`);
+						}
+						if (record.includes(inputValue)) {
+							//highlights the searched row
+							lastRow.css('background-color', '#DDC0B4');
+						}
+						$("tbody").append(lastRow);
+						$('#loader').hide();
+					}
+				})
+				.catch((err) => {
+					throw 'Error fetching records' + err;
+				})
```

Comment:
There is a lot of code duplication happening here, could you move this to a function and then rather use the function here and in the other branch of the `if`?